src/App.js

```diff
@@ -1,10 +1,51 @@
+import React, { useEffect, useState, useCallback } from 'react';
+
+import GlobalStyle from './styles/GlobalStyle';
+import { Main, Section } from './styles/Container';
+import { getLocalStorage, syncLocalStorage } from './components/TodoState';
+import { runToast, ToastComponent } from './components/toast/Toast';
+import InputTodo from './components/InputTodo';
+import TodoItemList from './components/TodoItemList';
 
 function App() {
-  return (
-    <div>
-      <h1>15기 화이팅~</h1>
-    </div>
-  );
+    const [list, setList] = useState(getLocalStorage);
+
+    // list 상태 변화할 때마다 로컬스토리지 동기화
+    useEffect(() => syncLocalStorage(list), [list]);
+
+    const toggleTodo = useCallback(
+        (id) => {
+            setList((list) => list.map((item) => (item.id === id ? { ...item, done: !item.done } : item)));
+        },
+        [list]
+    );
+
+    const removeTodo = useCallback(
+        (id) => {
+            setList((list) => list.filter((item) => item.id !== id));
+            runToast('삭제되었어요!');
+        },
+        [list]
+    );
+
+    return (
+        <>
+            <GlobalStyle />
+            <Main>
+                <Section>
+                    <h1>투두 리스트</h1>
+                    <InputTodo list={list} setList={setList} />
+                </Section>
+                <Section>
+                    <TodoItemList done={false} list={list} setList={setList} toggle={toggleTodo} remove={removeTodo}></TodoItemList>
+                </Section>
+                <Section>
+                    <TodoItemList done={true} list={list} setList={setList} toggle={toggleTodo} remove={removeTodo}></TodoItemList>
```

Comment:
자식 컴포넌트에서도 전달받은 props가 이벤트 핸들러라는 것을 명확하게 알 수 있도록 아래와같이 작성하는 편이 더 좋을 것 같습니다. 아니면 위에서 말씀드렸다시피 `handle + 대상 + 동작` 순서로 네이밍해 주시면 더 알아보기 쉽겠죠?

```suggestion
                    <TodoItemList done={false} list={list} setList={setList} toggleTodo={toggleTodo} removeTodo={removeTodo}></TodoItemList>
                </Section>
                <Section>
                    <TodoItemList done={true} list={list} setList={setList} toggle={toggleTodo} remove={removeTodo}></TodoItemList>
```